src/components/include/protocol_handler/session_observer.h

```diff
@@ -54,6 +54,58 @@ namespace protocol_handler {
  */
 enum { HASH_ID_NOT_SUPPORTED = 0, HASH_ID_WRONG = 0xFFFF0000 };
 
+/**
+ * @brief Struct with data containing attributes of starting session
+ **/
+struct StartingSessionContext {
+  transport_manager::ConnectionUID connection_id_;
+  uint8_t initial_session_id_;
+  uint32_t new_session_id_;
+  protocol_handler::ServiceType service_type_;
+  uint32_t hash_id_;
+  bool is_protected_;
+  bool is_new_service_;
+  bool is_ptu_required_;
+
+  /**
+   * @brief Constructor
+   */
+  StartingSessionContext()
```

Comment:
@AKalinich-Luxoft do we need empty constructor?

Reply:
@LuxoftAKutsan yes. Default constructed context is used in NotifyServiceStartedResult() for delayed initialization and also required in HandshakeHandler class in deprecated constructor as there is no actual context passed.